Adding log_warning option in tf.data.experimental.ignore_errors

stjohnso98 opened the pull request:
Resolves #38078.

Adds an option log_warning for ignore_errors which can be used to control whether error is logged to stderr.

stjohnso98 committed:
Adding log_warning option in ignore_errors that controls whether error is logged.

Signed-off-by: Steenu Johnson <steenu.johnson@students.iiserpune.ac.in>

jsimsa reviewed (dismissed):
@stjohnso98 thank you for the contribution and sorry for the review delay as I was out of office for an extended period of time

jsimsa commented on tensorflow/core/kernels/data/experimental/ignore_errors_dataset_op.cc:
```diff
@@ -15,6 +15,7 @@ limitations under the License.
 #include "tensorflow/core/framework/dataset.h"
 #include "tensorflow/core/framework/partial_tensor_shape.h"
 #include "tensorflow/core/framework/tensor.h"
+#include "tensorflow/core/platform/logging.h"
```
Internal tests are failing because this file does not directly depend on a module that provides this header file.

Please add the dependency "tensorflow/core/platform:logging" to this target.

stjohnso98 replied:
Thanks for your comment. I have added the dependency.

jsimsa commented on tensorflow/python/data/experimental/ops/error_ops.py:
```diff
@@ -20,10 +20,10 @@
 
 @tf_export("data.experimental.ignore_errors")
-def ignore_errors():
+def ignore_errors(log_warning=False):
```
You will need to regenerate the golden API files as your change is changing the public API.

You can do so by running:

```
$ bazel build tensorflow/tools/api/tests:api_compatibility_test
$ bazel-bin/tensorflow/tools/api/tests/api_compatibility_test --update_goldens True
```

jsimsa replied:
In addition, the docstring should be updated with an "Args:" section (see other source files for examples) that describe the purpose of the argument.

stjohnso98 replied:
Thanks for your comment. The docstring is updated with an "Args:" section. The golden API files is also updated.

gbaned commented:
@stjohnso98 Can you please check @jsimsa's comments and keep us posted ? Thanks!

stjohnso98 commented:
@jsimsa Thanks for your review. I am working on your comments and will update the PR soon. Thanks.

stjohnso98 committed:
Adding dependency tensorflow/core/platform:logging to ignore_errors_dataset_op.

Signed-off-by: Steenu Johnson <steenu.johnson@students.iiserpune.ac.in>

stjohnso98 committed:
Updating docstring with "Args:" section to describe log_warning argument.

Signed-off-by: Steenu Johnson <steenu.johnson@students.iiserpune.ac.in>

stjohnso98 committed:
Adding api_def_IgnoreErrorsDatasetV2.pbtxt file.

Signed-off-by: Steenu Johnson <steenu.johnson@students.iiserpune.ac.in>

stjohnso98 committed:
Updating golden API files.

Signed-off-by: Steenu Johnson <steenu.johnson@students.iiserpune.ac.in>

stjohnso98 commented:
@jsimsa The PR is updated with the requested changes. Can you please review again? Thanks.

jsimsa commented on tensorflow/core/kernels/data/experimental/BUILD:
```diff
@@ -248,6 +248,7 @@ tf_kernel_library(
     deps = [
         "//tensorflow/core:experimental_dataset_ops_op_lib",
         "//tensorflow/core:framework",
+	"//tensorflow/core/platform:logging",
```
fix indent

stjohnso98 replied:
Fixed.

jsimsa reviewed (dismissed):
one minor comment, otherwise LGTM

stjohnso98 committed:
Fixing indentation.

Signed-off-by: Steenu Johnson <steenu.johnson@students.iiserpune.ac.in>

alextp commented on tensorflow/core/ops/experimental_dataset_ops.cc:
```diff
@@ -441,6 +441,14 @@ REGISTER_OP("ExperimentalIgnoreErrorsDataset")
     .Attr("output_shapes: list(shape) >= 1")
     .SetShapeFn(shape_inference::ScalarShape);
 
+REGISTER_OP("IgnoreErrorsDatasetV2")
```
Can we make this a new attribute on IngoreErrorsDataset with a default value of False? That is technically not what we always recommend (we prefer inputs) but it feels that this is particularly unlikely to vary at runtime.

stjohnso98 replied:
Is this comment addressed to @tensorflow/api-owners?

alextp reviewed (commented):
For @tensorflow/api-owners

gbaned commented:
@stjohnso98  Can you please check @alextp's comments and keep us posted ? Thanks!

alextp commented:
It was a comment from the API owners, sorry for the confusion. We want you
to merge the new input as an attribute over the existing op to avoid the
need for an entirely new op.

On Thu, Jul 30, 2020 at 9:22 PM stjohnso98 <notifications@github.com> wrote:

> *@stjohnso98* commented on this pull request.
> ------------------------------
>
> In tensorflow/core/ops/experimental_dataset_ops.cc
> <https://github.com/tensorflow/tensorflow/pull/41336#discussion_r463397596>
> :
>
> > @@ -441,6 +441,14 @@ REGISTER_OP("ExperimentalIgnoreErrorsDataset")
>      .Attr("output_shapes: list(shape) >= 1")
>      .SetShapeFn(shape_inference::ScalarShape);
>
> +REGISTER_OP("IgnoreErrorsDatasetV2")
>
> Is this comment addressed to @tensorflow/api-owners?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/41336#discussion_r463397596>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAABHRO7T4OJ36TFM5VXYJTR6JBGNANCNFSM4OYMCQAQ>
> .
>


-- 
 - Alex

stjohnso98 commented:
@alextp Thanks for your comment. I am working on it.

gbaned commented:
@stjohnso98, Any update on this PR? Please. Thanks!

stjohnso98 committed:
Making the log_warning option as an attribute in IgnoreErrorsDataset instead of defining a new op.

Hence this commit also removes IgnoreErrorsDatasetV2 as it is no longer required.

Signed-off-by: Steenu Johnson <steenu.johnson@students.iiserpune.ac.in>

stjohnso98 commented:
@gbaned Apologies for the delay. The PR has been updated with the requested changes. @jsimsa @alextp Please review it again. Thanks.

alextp commented on tensorflow/python/data/experimental/ops/error_ops.py:
```diff
@@ -41,26 +40,30 @@ def ignore_errors():
     """See `Dataset.ignore_errors()` for details."""
     self._input_dataset = input_dataset
     variant_tensor = (
         gen_experimental_dataset_ops.ignore_errors_dataset(
```
This needs to be under a forward compatibility guard set three weeks from today. See https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/compat/compat.py

stjohnso98 replied:
Fixed.

stjohnso98 committed:
Adding three weeks forward compatibility condition to the op ignore_errors_dataset.

Signed-off-by: Steenu Johnson <steenu.johnson@students.iiserpune.ac.in>

stjohnso98 commented:
@alextp Thanks for your comment. I have made the requested change. Can you please review again? Thanks.
